doc/docs.md

```diff
@@ -2130,6 +2103,82 @@ In this example, the `can_register` method has a receiver of type `User` named `
 The convention is not to use receiver names like `self` or `this`,
 but a short, preferably one letter long, name.
 
+### Embedded structs
+
+V support embedded structs .
+
+Basically, V embedding should work the same way as Golang embedding .
+
+```v
+struct Size {
+mut:
+        width int
+        height int
+}
+
+fn (s &Size) area() int {
+        return s.width * s.height
+}
+
+struct Button {
+        Size
+        title string
+}
+```
+
+With embedding Button auto have field and method from `Size` do:
+
+```v oksyntax
+mut button := Button{
+	title: 'Click me'
+        height: 2
+}
+
+button.width = 3
+assert button.area() == 6
+assert button.Size.area() == 6
+print(button)
+```
+
+output :
+```v
+Button{
+    Size: Size{
+        width: 3
+        height: 2
+    }
+    title: 'Click me'
+}
+```
+
+Embedded struct like inheritance , struct will automatically have the fields and functions from embedded struct.
+
+Embedded struct is not inheritance , there is no parent-child relationship between embedded structs, and can't type cast.
```

Comment:
Hmm.. IMO it is subtle to assert that there is no parent-child relationship.
Changing the fields or methods of an embedded structure will naturally change the behaviour of the structure on the embedding side.

I think it's enough to say what embedded struct is that `button.widget.x` can simply be written as `button.x` as if you had defined a wrapper method.
And how it works if there is a method or field with the same name.

I think that's the simplest and most accurate way to describe it.

Reply:
As a side note, in my own projects I use embedding a lot, but I'm not a big fan of using it in the standard library.
If the standard library is designed from the outset to be inherited, as is the case with OOP languages, then it is carefully designed beforehand to avoid changing the behaviour of the classes later on.
However, the V standard library has not been designed in this way.
So when embedding is used, it is presumed that changes to methods will have a more unpredictable effect.

Reply:
I edited, as below

Embedded struct like inheritance, struct will automatically have the fields and methods from embedded struct.

Embedded struct is not inheritance, can't type cast between structs and embedded structs.

If you want to access embedded structs, write something like `button.Size`.

Conceptually, embedded struct similar [mixin](https://en.wikipedia.org/wiki/Mixin) in OOP, not base class.

An embedded structs is responsible for implementing a small point and exposing a few methods, just like Lego blocks. 

Don't bother to elaborate a bulky base class with huge methods. Needn't import a forest for a banana.

>  The problem with object-oriented languages is they’ve got all this implicit environment that they carry around with them. You wanted a banana but what you got was a gorilla holding the banana and the entire jungle.
 
—— Joe Armstrong, creator of Erlang progamming language